editor/src/core/model/performance-scripts.tsx

```diff
@@ -331,6 +338,174 @@ export function useTriggerSelectionPerformanceTest(): () => void {
+      position: 'absolute',
+      left:
+        childMetadata.globalFrame.x -
+        childMetadata.specialSizeMeasurements.coordinateSystemBounds.x,
+      top:
+        childMetadata.globalFrame.y -
+        childMetadata.specialSizeMeasurements.coordinateSystemBounds.y,
+      width: childMetadata.globalFrame.width,
+      height: childMetadata.globalFrame.height,
+    }
+
+    // Determine where the events should be fired.
+    const controlsContainerElement = forceNotNull(
+      'Container controls element should exist.',
+      document.getElementById(CanvasControlsContainerID),
+    )
+    const canvasContainerElement = forceNotNull(
+      'Canvas container element should exist.',
+      document.getElementById(CanvasContainerID),
+    )
+    const canvasContainerBounds = canvasContainerElement.getBoundingClientRect()
+    const navigatorElement = forceNotNull(
+      'Navigator element should exist.',
+      document.getElementById(NavigatorContainerId),
+    )
+    const navigatorBounds = navigatorElement.getBoundingClientRect()
+
+    const targetElement = forceNotNull(
+      'Target element should exist.',
+      document.querySelector(`*[data-paths~="${EP.toString(childTargetPath)}"]`),
+    )
+    const originalTargetBounds = targetElement.getBoundingClientRect()
+    const leftToTarget =
+      canvasContainerBounds.left + navigatorBounds.width - originalTargetBounds.left + 100
+    const topToTarget = canvasContainerBounds.top - originalTargetBounds.top + 100
+    await dispatch(
+      [CanvasActions.positionCanvas(canvasPoint({ x: leftToTarget, y: topToTarget }))],
+      'everyone',
+    ).entireUpdateFinished
+    const targetBounds = targetElement.getBoundingClientRect()
+
+    let framesPassed = 0
+    async function step() {
+      // Make the div inside the target absolute positioned and ensure it is selected.
+      await dispatch(
+        [
+          selectComponents([childTargetPath!], false),
+          setProp_UNSAFE(
+            childTargetPath!,
+            PP.create(['style']),
+            jsxAttributeValue(styleValue, emptyComments),
+          ),
+        ],
+        'everyone',
+      ).entireUpdateFinished
+      performance.mark(`absolute_move_step_${framesPassed}`)
+
+      // Move it down and to the right.
+      controlsContainerElement.dispatchEvent(
+        new MouseEvent('mousedown', {
+          detail: 1,
+          bubbles: true,
+          cancelable: true,
+          metaKey: false,
+          clientX: targetBounds.left + 5,
+          clientY: targetBounds.top + 5,
+          buttons: 1,
+        }),
+      )
+      controlsContainerElement.dispatchEvent(
+        new MouseEvent('mousemove', {
+          detail: 1,
+          bubbles: true,
+          cancelable: true,
+          metaKey: false,
+          clientX: targetBounds.left + 35,
+          clientY: targetBounds.top + 45,
+          buttons: 1,
+        }),
+      )
+      controlsContainerElement.dispatchEvent(
```

Comment:
This current test will skew extremely towards mouse up performance (which right now is _very_ expensive because of the inspector remount), and thus it will actually measure the strategy lifecycle more than the move speed itself. I'd be tempted to rename this to Something Something Strategy Lifecycle.

And for the Absolute Move Strategy Performance itself I'd add a second test in which only the mouse move is inside the loop, so we have a number for continuous scrolling performance.

Reply:
Stepped mouse move added in 3bbedfa25.

Reply:
9060f4687 captures both the entire interaction time and just the mouse move segments of the absolute move interaction.